#2502 Fix CoreData concurrency issues in RKManagedObjectRequestOperation & RKManagedObjectResponseMapperOperation

rubnov opened the pull request.

valeriomazzeo committed:
prepare 0.27.1 release

valeriomazzeo committed:
fixed delete database issue

valeriomazzeo committed:
updated project settings
updated travis file to spin an xcode8 image

Commit:
Shouldn’t validate content type if there is no content

Commit:
Removed no longer necessary comment

Commit:
#2502 accommodate for CoreData concurrency in RKErrorFromMappingResult

Commit:
add RKObjectRequestOperation::didFinish with override in RKManagedObjectRequestOperation to accommodate for CoreData concurrencty

samkrishna committed:
Fixed test failure by changing URL to a new invalid address

valeriomazzeo committed:
updated dependencies

valeriomazzeo committed:
rvm 2.4.1

valeriomazzeo committed:
updated cocoapods and specta dependencies

valeriomazzeo committed:
generic simulator

valeriomazzeo committed:
fix bundler cache

valeriomazzeo committed:
10.3.1

valeriomazzeo committed:
consolidated use of localhost

valeriomazzeo committed:
commented test

valeriomazzeo committed:
fixed issue with removeAllCachedResponse being slow on the simulator
since xcode8 clean action is not supported

valeriomazzeo committed:
removed xctool that is not supported for building anymore
separated linting and tests into two scripts so we can use travis_wait and still have some output during the build
fixed bundler cache

valeriomazzeo committed:
replaced while with expecta

valeriomazzeo committed:
fixed documentation warnings

valeriomazzeo committed:
split build into 2 jobs

valeriomazzeo commented:
@rubnov could you please update from `development` and make sure tests are green?

Also, a unit test that reproduces the problem that has been fixed would be greatly appreciated.

rubnov commented:
@valeriomazzeo done. Didn't have time to write specific unit tests to the fixes I've made though. I doubt I'll get to it any time soon. Sorry.

valeriomazzeo committed:
Issue 2508: __NSCFBoolean private API (#2510)

valeriomazzeo committed:
bump version 0.27.2

SimonMcFarlane committed:
Changed RKStringTokenizer to allow tokenization of symbol characters (#2516)

* Changed the `RKStringTokenizer` method `- (NSSet *)tokenize:(NSString *)string` to use `kCFStringTokenizerUnitWordBoundary` instead of `kCFStringTokenizerUnitWord`. This then allows tokens to be created for symbols such as '@', '$' and '#' and therefore making them searchable.

As a result of using `kCFStringTokenizerUnitWordBoundary` instead of `kCFStringTokenizerUnitWord`, spaces are detected and a token created. As we don't want to be returning results that include a space, this token (if it exists) is removed before the tokens are returned.

Unit tests added to cover both the creating of tokens from strings that contain symbols and searching for managed objects that contain symbols.

* Core formatting - removed tabs from empty lines in RKStringTokenizer.

* Fixed assert mistake in RKSearchTest unit test `testSearchingForManagedObjectsWithSymbols`.

slxl committed:
Method signature update

To fix Xcode 9 "The block declaration is not a prototype" warning

gusennan committed:
Remove duplicate word in README

Change "to to" to "to"

br-tyler-milner committed:
Added unit test for making sure the URLRequest is returned in the completion block of the 'setImageWithURLRequest:placeholderImage:success:failure:' method on the AFNetworking UIImageView extension.

br-tyler-milner committed:
Adopted library testing syntax and stubbed out a couple of extra tests. The test is still failing at this point since I haven't yet fixed the bug.

br-tyler-milner committed:
Fixes #2529. Fixed bug where the NSURLRequest passed into 'setImageWithURLRequest:placeholderImage:success:failure:' wasn't being returned in the 'success' completion block when the image was fetched from the cache. The unit test is passing now.

br-tyler-milner committed:
Support for previous commit. Extra bonus for #2529. Implemented tests for the rest of the parameters in the 'success' block.

br-tyler-milner committed:
Fixes #2531. Fixed namespace collision with the official AFNetworking library's implementation of the UIImageView helper category for setting an image from a URL. This will be a breaking change, but should help make people aware of potential unwanted behavior if their project includes both AFNetworking and RestKit.

Commit:
Perform access to NSManagedObjectContext's properties only on its queue

Commit:
Add test that represents the issue

valeriomazzeo committed:
Fixed xcode 9 warnings

valeriomazzeo committed:
fixed typo

yauhenidrobau committed:
- ios 12 crash entity is not visible to managedObject context

valeriomazzeo committed:
bump version 0.27.3

rubnov committed:
Merge branch 'development' of https://github.com/RestKit/RestKit into development

rubnov committed:
Fix `Use of private header from outside its module: netinet6/in6.h`
- removed offending `#import <netinet6/in6.h>` from AFRKHTTPClient.m (already included in <netinet/in.h>)
- increment ruby-version to 3.2.2
- increment iOS min target version to 16.6, min Mac OS version to 11.5
- increment dependency versions in podfile
